libs/design/src/molecules/media-gallery/registry/media-gallery.registry.spec.ts

```diff
@@ -0,0 +1,137 @@
+import { ChangeDetectorRef } from '@angular/core';
+import {
+  waitForAsync,
+  TestBed,
+} from '@angular/core/testing';
+
+import { DaffMediaGalleryComponent } from '../media-gallery.component';
+import { DaffThumbnailDirective } from '../thumbnail/thumbnail.directive';
+import { DaffMediaGalleryRegistry } from './media-gallery.registry';
+
+describe('DaffMediaGalleryRegistry', () => {
+  let service: DaffMediaGalleryRegistry;
+  let mockGalleryAlreadyAdded;
+  let mockThumbnailAlreadyAdded;
+
+  beforeEach(waitForAsync(() => {
+    TestBed.configureTestingModule({
+      providers: [
+        DaffMediaGalleryRegistry,
+        ChangeDetectorRef,
+      ],
+    });
+    service = TestBed.inject(DaffMediaGalleryRegistry);
+
+    mockGalleryAlreadyAdded = new DaffMediaGalleryComponent();
+    mockThumbnailAlreadyAdded = new DaffThumbnailDirective(null, jasmine.createSpyObj('ChangeDetectorRef', ['markForCheck']), null, mockGalleryAlreadyAdded);
+    service.add(mockGalleryAlreadyAdded, mockThumbnailAlreadyAdded);
+  }));
+
+  it('should be created', () => {
+    expect(service).toBeTruthy();
+  });
+
+  describe('add', () => {
+
+    describe('when the gallery given already exists in the registry', () => {
+
+      it('should add the thumbnail to the gallery given', () => {
+        const newThumbnail = new DaffThumbnailDirective(null, jasmine.createSpyObj('ChangeDetectorRef', ['markForCheck']), null, mockGalleryAlreadyAdded);
```

Comment:
Given that you don't call any functions of the thumbnail, this can be simply replaced with a mock instead of a real object with mocked deps?

Reply:
done